inlang/source-code/sdk/src/persistence/storeApi.ts

```diff
@@ -0,0 +1,16 @@
+import type * as V2 from "../v2/types.js"
+
+export type StoreApi = {
+	messageBundles: Query<V2.MessageBundle>
+}
+
+/**
+ * WIP template for async V2 crud interfaces
+ * E.g. `await project.messageBundles.get({ id: "..." })`
+ **/
+export interface Query<T> {
+	get: (args: { id: string }) => Promise<T | undefined>
+	set: (args: { data: T }) => Promise<void>
```

Comment:
The api of rxdb looks pretty clean and carefully drafted [see](https://rxdb.info/rx-query.html#update) I use this as an inspiration

Reply:
Agree. I would just copy the mango query builder and be done with discussions how our query Api should look like.

Reply:
if the argument is that a mango query builder would take 1-2 weeks longer to implement, i would invest those 1-2 weeks:

1. having a basic query builder now will ship the SDK faster but slow down apps (aka no speed advantage)
2. breaking change in the future because of argument 1

Reply:
Thanks, that's helpful guidance.

The query builder api needs to live in a layer above this single entity, low-level store api, so that it can include other entities like lint reports. I will rename this api to Store<T> (instead of Query<T>) to make this clearer.

Reply:
Ah, okay Store instead of Query makes sense.

> the implementation can be non-performance optimized. as long as the API is set, apps have an easy time querying now and we avoid a breaking change

Addding to this remark: The query API doesn't even need to be complete. Functionality like filters (`$gte`) can be added incrementally.